### PR TITLE
Akka-free event-sourced persistence

### PR DESCRIPTION
**Pull request by dfakhritdinov**

### CODE REVIEW

File: eventsourcing-persistence/src/main/scala/com/evolution/akkaeffect/eventsopircing/persistence/EventSourcedStore.scala [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    * @see [[com.evolutiongaming.akkaeffect.persistence.EventSourcedId]]
+    * @see [[akka.persistence.PersistentActor.persistenceId]]
+    */
+  final case class Id(value: String) extends AnyVal
+
+  /**
+    * Snapshot lookup criteria
+    * @see [[akka.persistence.SnapshotSelectionCriteria]]
+    */
+  final case class Criteria(maxSequenceNr: Long = Long.MaxValue,
```
Comment (t3hnar) [inline eventsourcing-persistence/src/main/scala/com/evolution/akkaeffect/eventsopircing/persistence/EventSourcedStore.scala:36]: not sure we need to support this

File: persistence/src/main/scala/com/evolutiongaming/akkaeffect/persistence/EventSourcedStoreOf.scala [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    * first events will be available for [[Stream#foldWhileM]] while tail still loaded by [[AsyncRecovery]]
+    *
+    * @param snapshotStore Akka Persistence snapshot (plugin)
+    * @param journal Akka Persistence journal (plugin)
+    * @tparam F effect
+    * @tparam S snapshot
+    * @tparam E event
+    * @return resource of [[EventSourcedStore]]
+    */
+  def fromAkka[F[_]: Async: ToTry, S, E](snapshotStore: SnapshotStore,
```
Comment (t3hnar) [inline persistence/src/main/scala/com/evolutiongaming/akkaeffect/persistence/EventSourcedStoreOf.scala:39]: overall I'm more interested in actual implementation that meant to replace `PersistentActor` :)

File: eventsourcing-persistence/src/main/scala/com/evolution/akkaeffect/eventsopircing/persistence/EventSourcedStore.scala (hunk 1 above)
Comment (dfakhritdinov) [reply eventsourcing-persistence/src/main/scala/com/evolution/akkaeffect/eventsopircing/persistence/EventSourcedStore.scala:36]: I was thinking about always using default value but still provided it via the API. Will drop it 👍

File: persistence/src/main/scala/com/evolutiongaming/akkaeffect/persistence/EventSourcedStoreOf.scala (hunk 2 above)
Comment (dfakhritdinov) [reply persistence/src/main/scala/com/evolutiongaming/akkaeffect/persistence/EventSourcedStoreOf.scala]: Please take a look on basic part of persistent actor replacement: `com.evolutiongaming.akkaeffect.persistence.EventSourcedActorOf`

**Comment by dfakhritdinov** [comment]
Okay, now I clearly see that my attempt of loading journaller/snapshotter plugins avoiding registration in akka system is a mistake. Instead, I will create an actor capable of interacting with properly registered journaller/snapshotter plugins ...

**Pull request closed without merging**